### DISCUSSION TITLE
Tycho 3.0 requires new Maven based JUnit bundles, even on older targets

### DISCUSSION THREAD

**Discussion by Bananeweizen** [eclipse-tycho/tycho#1457] [Q&A] [answered]
Upgrading multiple builds with Tycho 2.7.5 to 3.0 on company projects with older target platforms leads to missing dependency errors mentioning the new 2022-09 Maven based JUnit Jupiter bundles. Those are not available anywhere in older targets and surely not mentioned in any of my files. Does this mean Tycho 3.0 cannot be used on target platforms older than 2022-09?

```
[INFO] Resolving target definition file:/C:/some.feature/../../releng/my.target/2021-06.target for environments=[win32/win32/x86_64], include source mode=honor, execution environment=StandardEEResolutionHints [executionEnvironment=OSGi profile 'JavaSE-17' { source level: 17, target level: 17}], remote p2 repository options=org.eclipse.tycho.p2.remote.RemoteAgent@3203a4ae...
[INFO] {osgi.os=win32, osgi.ws=win32, org.eclipse.update.install.features=true, osgi.arch=x86_64}
[ERROR] Cannot resolve project dependencies:
[ERROR]   You requested to install 'org.eclipse.equinox.p2.iu; junit-jupiter-api [5.0.0,6.0.0)' but it could not be found
```

**Comment by laeubi** [comment, marked as answer]
You project most probably requesting a JUNIT5 classpath container. As those have changed with 2022-09 Tycho has adapted accordingly. You can add the JUnit dependencies as maven location in your target, or upgrade your project to use a newer eclipse target.

**Comment by lppedd** [comment]
> You can add the JUnit dependencies as maven location in your target

Thanks! Basically we should add JUnit 5 as dependency even if we're not really using it, that's the meaning?

**Reply by laeubi** [reply]
If you are not using JUNIT5 you should not add a JUNIT5 classpath container or anything else related to JUnit 5.

If you does, the you are actually "really using it" :-)

**Comment by Bananeweizen** [comment]
Just as a note: I've meanwhile used that solution in one project successfully. The target platform there includes both the old JUnit 5.8.1 bundles (because SWTBot requires them) and the new renamed 5.9.x based (because Tycho 3.0 requires them).

**Reply by laeubi** [reply]
Thanks for the feedback, do you like to mark the answer just in case other stumble about the same problem?

**Reply by Bananeweizen** [reply]
Done. Unfortunately the part "or upgrade your target" is often not that easy in bigger environments, because some other plugins may still not work with the new version. See the above example of SWTBot, which requires the old bundle name.

**Reply by lppedd** [reply]
> and the new renamed 5.9.x based

What do you mean with renamed?

**Reply by Bananeweizen** [reply]
https://github.com/eclipse-platform/eclipse.platform.common/blob/master/bundles/org.eclipse.platform.doc.isv/porting/4.25/incompatibilities.html#L36. junit bundles in 2022-06 have different names than the previous same libraries from Orbit.

**Reply by lppedd** [reply]
@Bananeweizen ahh ok, I thought you had to somehow rename/alias those dependencies and I got scared lol

**Reply by lppedd** [reply]
@Bananeweizen sorry to bother again. If you have time, could you share a snippet of your target platform? I'm going crazy.
